### PR TITLE
We should also automatically test the examples within the documentation

### PR DESCRIPTION
**Pull request by AlexanderWillner**

### CODE REVIEW

File: things/database.py [hunk 1]
```diff
@@ -754,12 +754,8 @@ def make_search_filter(query: str) -> str:
 
     Example:
     --------
-    >>> make_search_filter('dinner')
-    'AND (
-        TASK.title LIKE "%dinner%"
-        OR TASK.notes LIKE "%dinner%"
-        OR AREA.title LIKE "%dinner%"
-    )'
+    >>> make_search_filter('dinner') #doctest: +REPORT_NDIFF
```
Comment (mikez) [inline things/database.py:757]: The docstring are for humans to read, not machines. The "#doctest" makes it harder to read.

**Comment by mikez** [comment]
I'd like you to hear my preference: Human readability is always the first priority. Think of somebody who's never seen the API or this code before. Examples serve to bring understanding. They don't need to match the test database. In fact, I used the example database that Things use in their demos. See the Readme.md examples. :)

File: things/database.py [hunk 2]
```diff
@@ -655,7 +655,7 @@ def make_date_filter(date_column: str, value) -> str:
     'AND start_date IS NULL'
 
     >>> make_date_filter('start_date', 'future')
-    "AND datetime(start_date, 'unixepoch') > datetime('now')"
+    "AND date(start_date, 'unixepoch', 'localtime') > date('now', 'localtime')"
```
[the remark this answers is not in the document]
Comment (AlexanderWillner) [reply things/database.py:658]: catched automatically by the tool, not me ;)

File: things/database.py (hunk 1 above)
Comment (AlexanderWillner) [reply things/database.py:757]: Wasn't supposed to be committed. Will remove it in a new pull request.

File: things/api.py [hunk 3]
```diff
@@ -613,12 +613,12 @@ def completed(**kwargs):
 
     Examples
     --------
-    >>> things.completed(count_only=True)
-    14
-    >>> things.completed(type='project', count_only=True)
-    4
+    >>> import things; things.completed(count_only=True)
+    10
```
Comment (mikez) [inline things/api.py:617]: "import things" here decreases readability. See how pandas and numpy do it.

**Pull request merged**